src/authtoken/authentication.py

```diff
@@ -51,3 +70,46 @@ def authenticate_credentials(self, key):
 
     def authenticate_header(self, request):
         return 'Token'
+
+
+class FooCardAuthentication(authentication.BaseAuthentication):
+    """
+    JSON Web Token authentication.
+    Acts identically to token based authentication, as seen from the outside.
+    Clients should authenticate by passing the token key in the "Authorization"
+    HTTP header, prepended with the string "CardToken ". For example:
+        Authorization: CardToken: 401f7ac837da42b97f613d789819ff93537bee6a
+    """
+    algorithm = 'HS256'
+    message = 'Invalid token'
+    model = Account
+
+    def authenticate(self, request):
+        auth = get_authorization_header(request).split()
+        if not auth or auth[0].lower() != b'cardtoken':
+            return None
+
+        validate_header(auth)
+        try:
+            payload = jwt.decode(
+                auth[1].decode(),
+                settings.SECRET_WEBTOKEN,
+                algorithm=self.algorithm
+            )
+        except (InvalidKeyError, InvalidTokenError):
+            msg = _('{0} header'.format(self.message))
+            raise exceptions.AuthenticationFailed(msg)
+
+        return self.authenticate_credentials(payload)
+
+    def authenticate_credentials(self, payload):
+        account_id = payload.get('account_id')
+        try:
+            account = self.model.objects.get(pk=account_id)
```

Comment:
We can utilize the `get_card` function from the api instead. And since we are posting the card number to get a valid token, we place that number in the payload. Might actually make a little bit more sense as well.